File: src/hardware_monitors/laser_monitors/laser_device_monitor.cpp
#include "hardware_monitors/laser_monitors/laser_device_monitor.h"

namespace monitoring
{
    LaserDeviceMonitor::LaserDeviceMonitor(config::MonitorModeConfig config_params)
        : MonitorBase(config_params)
    {
        for (auto mapping : config_params.mappings)
        {
            // each input mapping has a single device name, so we just take the first element
            device_names_.push_back(mapping.inputs[0]);
            device_status_names_.push_back(mapping.outputs[0].name);
        }
    }

    Json::Value LaserDeviceMonitor::getStatus()
    {
        Json::Value status_msg = getStatusMessageTemplate();
        status_msg["monitorName"] = config_params_.name;
        for (size_t i=0; i<device_names_.size(); i++)
        {
            status_msg["healthStatus"][device_status_names_[i]] = boost::filesystem::exists(device_names_[i]);
        }
        return status_msg;
    }
}

File: component_monitoring/monitors/hardware/laser/laser_device_monitor.py
from component_monitoring.monitor_base import MonitorBase

class LaserDeviceMonitor(MonitorBase):
    def __init__(self, config_params):
        super(LaserDeviceMonitor, self).__init__(config_params)
        self.device_names = list()
        self.device_status_names = list()
        for mapping in config_params.mappings:
            self.device_names.append(mapping.inputs[0])
            self.device_status_names.append(mapping.outputs[0].name)

    def get_status(self):
        status_msg = self.get_status_message_template()
        status_msg['monitorName'] = self.config_params.name
        status_msg['healthStatus'] = dict()
        for i, device_name in enumerate(self.device_names):
            status_msg['healthStatus'][self.device_status_names[i]] = False
        return status_msg
